content-moderation-evaluation/Evaluating-Amazon-Rekognition-Content-Moderation-Service.ipynb

```diff
@@ -0,0 +1,503 @@
+    "!pip install sagemaker --upgrade"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In order to start, it's necessary to create a bucket where to host evaluation dataset, then set proper values for following variables."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import itertools\n",
+    "import json\n",
+    "import time\n",
+    "import boto3\n",
+    "import sagemaker\n",
+    "\n",
+    "BUCKET = '<YOUR S3 BUCKET NAME>'          # S3 bucket holds your evaluation dataset\n",
+    "FILE_PREFIX = '<IMAGE PREFIX>'              # The prefix for your evaluation dataset\n",
+    "EXP_NAME = '<JOB PREFIX>'                   # S3 prefix for SageMaker Ground Truth labeling job, please do not add trailing \"/\"\n",
+    "INPUT_MANIFEST = '<INPUT FILENAME>'         # Input manifest filename for SageMaker Ground Truth labeling job e.g. input.manifest\n",
+    "OUTPUT_MANIFEST = '<OUTPUT FILENAME>'       # Output manifest filename for SageMaker Ground Truth labeling job e.g. output.manifest"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Make sure the bucket is in the same region as this notebook."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "role = sagemaker.get_execution_role()\n",
+    "region = boto3.session.Session().region_name\n",
+    "s3 = boto3.client(\"s3\")\n",
+    "bucket_region = s3.head_bucket(Bucket=BUCKET)[\"ResponseMetadata\"][\"HTTPHeaders\"][\n",
+    "    \"x-amz-bucket-region\"\n",
+    "]\n",
+    "assert (\n",
+    "    bucket_region == region\n",
+    "), \"You S3 bucket {} and this notebook need to be in the same region.\".format(BUCKET)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 2: Use Amazon SageMaker Ground Truth service to assign ground truth moderation labels to the evaluation dataset. <a id=\"step2\"></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 1. Create input manifest file for Ground Truth job\n",
+    "When listing objects in s3 bucket, the list_objects_v2 API by default return up to 1000 objects. If you have more than 1000 iamges in your bucket, we suggest to set [MaxKeys](https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectsV2.html#AmazonS3-ListObjectsV2-response-MaxKeys) to be equal or larger than the number of images, you can also add check for [IsTruncated](https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectsV2.html#AmazonS3-ListObjectsV2-response-IsTruncated) in the response and then potentially adding a loop for pagination, please refer to [ListObjectsV2 documentation](https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectsV2.html) for more details. "
```

Comment:
Setting MaxKeys to more than 1000 will not work, you need to adapt the code to perform the loop as i suggested in my code sample or inform the users that this example will not work with more than 1000.

You also have a typo here `iamges`

Reply:
Understood, I've updated code as suggested and corrected markdown.